src/materializer_vnode.erl

```diff
@@ -335,8 +272,65 @@ terminate(_Reason, _State=#mat_state{ops_cache=OpsCache, snapshot_cache=Snapshot
 
 %%---------------- Internal Functions -------------------%%
 
--spec internal_store_ss(key(), #materialized_snapshot{}, snapshot_time(), boolean(), #mat_state{}) -> boolean().
-internal_store_ss(Key, Snapshot = #materialized_snapshot{last_op_id = NewOpId}, CommitTime, ShouldGc, State = #mat_state{snapshot_cache=SnapshotCache}) ->
+-spec get_cache_name(non_neg_integer(), atom()) -> atom().
+get_cache_name(Partition, Base) ->
+    list_to_atom(atom_to_list(Base) ++ "-" ++ integer_to_list(Partition)).
+
+-spec load_from_log_to_tables(partition_id(), #state{}) -> ok | {error, reason()}.
+load_from_log_to_tables(Partition, State) ->
+    LogId = [Partition],
+    Node = {Partition, log_utilities:get_my_node(Partition)},
+    loop_until_loaded(Node, LogId, start, dict:new(), State).
+
+-spec loop_until_loaded({partition_id(), node()},
+            log_id(),
+            start | disk_log:continuation(),
+            dict:dict(txid(), [any_log_payload()]),
+            #state{}) ->
+                   ok | {error, reason()}.
+loop_until_loaded(Node, LogId, Continuation, Ops, State) ->
+    case logging_vnode:get_all(Node, LogId, Continuation, Ops) of
+        {error, Reason} ->
+            {error, Reason};
+        {NewContinuation, NewOps, OpsDict} ->
+            load_ops(OpsDict, State),
+            loop_until_loaded(Node, LogId, NewContinuation, NewOps, State);
+        {eof, OpsDict} ->
+            load_ops(OpsDict, State),
+            ok
+    end.
+
+-spec load_ops(dict:dict(key(), [{non_neg_integer(), clocksi_payload()}]), #state{}) -> true.
```

Comment:
The honest return type would be `true | ok`, because `lists:foreach` returns `ok`